### PR TITLE
cli: Add exit-codes to install command

### LINKED ISSUE #361: Add error codes to elemental-cli
Add errors to following parts and its dependencies:

 - [x] #362
 - [x] #363
 - [ ] #364
 - [x] #365
 - [x] #366
 - [x] #367
 - [x] #368
 - [x] #369
 - [x] #370

Example Exit codes PR: https://github.com/rancher/elemental-cli/pull/355

### LINKED ISSUE #365: Add exit codes to install

### PR DESCRIPTION
**Pull request by frelon**
Fixes #365

Signed-off-by: Fredrik Lönnegren <fredrik.lonnegren@suse.com>

### CODE REVIEW

File: cmd/install.go [hunk 1]
```diff
@@ -70,24 +72,20 @@ func NewInstallCmd(root *cobra.Command, addCheckRoot bool) *cobra.Command {
 			spec, err := config.ReadInstallSpec(cfg, cmd.Flags())
 			if err != nil {
 				cfg.Logger.Errorf("invalid install command setup %v", err)
-				return err
+				return elementalError.NewFromError(err, elementalError.ReadingSpecConfig)
```
Comment (Itxaka) [inline cmd/install.go:75]: uh, interesting that in the install we DO return an error if we cant load the config... :thinking:

File: pkg/action/install.go [hunk 2]
```diff
@@ -269,13 +270,5 @@ func (i InstallAction) Run() (err error) {
...
-	// Reboot, poweroff or nothing
-	if i.cfg.Reboot {
-		i.cfg.Logger.Infof("Rebooting in 5 seconds")
-		return utils.Reboot(i.cfg.Runner, 5)
-	} else if i.cfg.PowerOff {
-		i.cfg.Logger.Infof("Shutting down in 5 seconds")
-		return utils.Shutdown(i.cfg.Runner, 5)
-	}
-	return err
+	return PowerAction(i.cfg)
```
Comment (Itxaka) [inline pkg/action/install.go:273]: nice

**Review by Itxaka — APPROVED** [review]
very nice!

File: pkg/action/install.go (hunk 2 above)
Comment (frelon) [reply pkg/action/install.go:273]: :muscle:

File: cmd/install.go (hunk 1 above)
Comment (frelon) [reply cmd/install.go:75]: aha! but it looks like it always does that in the cmd/* and then not returns in ReadRunConfig/ReadUpdateSpec.. I will remove the error-return I added in the last PR just to be consistent!

**Pull request merged**